lib/model/folderscanner.go

```diff
@@ -50,10 +50,32 @@ func (f *folderScanner) Scan(subdirs []string) error {
 		subdirs: subdirs,
 		err:     make(chan error),
 	}
+	l.Infoln("DEBUG scan requests channel: %p", f.now)
 	f.now <- req
 	return <-req.err
 }
 
 func (f *folderScanner) Delay(next time.Duration) {
 	f.delay <- next
 }
+
+type folderScannerFactory struct {
+	hasSingleGlobalFolderScanner bool
+	globalScanRequests           chan rescanRequest
+}
+
+func newFoldeScannerFactory(hasSingleGlobalFolderScanner bool) folderScannerFactory {
+	return folderScannerFactory{
+		hasSingleGlobalFolderScanner: hasSingleGlobalFolderScanner,
+		globalScanRequests:           make(chan rescanRequest),
+	}
+}
+
+func (fsf *folderScannerFactory) CreateOrGetSingleGlobalOrNewChannel() chan rescanRequest {
```

Comment:
Srsly?

https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition

Why do we even need a factory.
I'd use a semaphore, and store how many concurrent scans we allow.

Reply:
Also, I am not sure this does what you expect. I suspect this causes any random folder to be scanned.

Reply:
LOL, needed to read and ROFL.
I dont play often with semaphores ... thx for the hint.

Still laughing ... GREAT!

Reply:
Why should it scan randomly?

Reply:
Because every folder is reading and sending to the same channel

Reply:
aaah ... I see ... in the folder impl's. ok. 

```
case req := <-f.scan.now:
			req.err <- f.scanSubdirs(req.subdirs)
``` 

Silly me. :|